python-package/lightgbm/dask.py

```diff
@@ -169,24 +180,33 @@ def _split_to_parts(data, is_matrix):
     return parts
 
 
-def _train(client, data, label, params, model_factory, sample_weight=None, group=None, **kwargs):
+def _train(
+    client: Client,
+    data: _DaskCollection,
```

Comment:
Is it OK that `_DaskCollection = Union[dask_Array, dask_Frame]` but `data` can be only `dask DataFrame`, but not `dask Series`?

Reply:
you're absolutely right, thank you. I'll make a new type `_DaskArrayLike` for all of these

Reply:
I decided on `_DaskMatrixLike` to reduce confusion. This change also means we don't need `dask_Frame` any more.

Added in https://github.com/microsoft/LightGBM/pull/3866/commits/8b19e04b05f83a38f5213c1103040f49d0a20988